api/src/main/java/com/epam/pipeline/manager/preprocessing/NgsPreprocessingManager.java

```diff
@@ -0,0 +1,347 @@
+                                folderId)));
+        final DataStorageLink dataFolderPath = fetchDataFolder(folderMetadata);
+
+        final MetadataEntity machineRunMetadata = fetchMachineRunMetadataEntity(dataFolderPath, machineRunId);
+
+        final String sampleMetadataClassName = preferenceManager.getStringPreference(
+                SystemPreferences.PREPROCESSING_SAMPLE_CLASS.getKey());
+        final MetadataClass sampleMetadataClass = metadataEntityManager.loadClass(sampleMetadataClassName);
+
+        final String machineRunToSampleColumn = preferenceManager.getStringPreference(
+                SystemPreferences.PREPROCESSING_MACHINE_RUN_TO_SAMPLE_COLUMN.getKey());
+
+        Optional.ofNullable(machineRunMetadata.getData().get(machineRunToSampleColumn))
+                .ifPresent(value -> {
+                    if (EntityTypeField.isArrayType(value.getType())) {
+                        final List<String> samples = JsonMapper
+                                .parseData(value.getValue(), new TypeReference<List<String>>() {});
+                        for (String sample : samples) {
+                            final MetadataEntity sampleMetadataEntity = metadataEntityManager
+                                    .loadByExternalId(sample, sampleMetadataClass.getName(), folderId);
+                            metadataEntityManager.deleteMetadataEntity(sampleMetadataEntity.getId());
+                        }
+                    }
+                });
+
+        metadataEntityManager.deleteMetadataItemKey(machineRunMetadata.getId(), machineRunToSampleColumn);
+
+        final String sampleSheetFilePath = Paths.get(
+                dataFolderPath.getPath(),
+                machineRunMetadata.getExternalId(),
+                preferenceManager.getStringPreference(SystemPreferences.PREPROCESSING_SAMPLESHEET_FILE_NAME.getKey())
+        ).toString();
+        final AbstractDataStorage dataStorage = storageManager.load(dataFolderPath.getDataStorageId());
+        if (checkPathExistence(dataStorage.getId(), sampleSheetFilePath)) {
+            final UpdateDataStorageItemVO sampleSheetItem = new UpdateDataStorageItemVO();
+            sampleSheetItem.setPath(sampleSheetFilePath);
+            sampleSheetItem.setType(DataStorageItemType.File);
+            storageManager.deleteDataStorageItems(
+                    dataFolderPath.getDataStorageId(),
+                    Collections.singletonList(sampleSheetItem),
+                    dataStorage.isVersioningEnabled()
+            );
+        }
+    }
+
+    private void linkSamplesToMachineRun(final Long folderId, final MetadataEntity machineRunMetadataEntity,
+                                         final String sampleMetadataClass, final List<MetadataEntityVO> samples,
+                                         final String machineRunToSampleColumn) {
+        machineRunMetadataEntity.getData().put(
```

Comment:
Can we handle situation when some entities shall be deleted?

Reply:
do you mean situation when we need to delete old one entities? Current implementation design to totally overwrite existing entities for that purpose `deleteSampleSheet` is calling before link a new ones